src/site/content/en/mini-apps/mini-app-open-source-projects/index.md

```diff
@@ -0,0 +1,295 @@
+[Preact](https://github.com/wechat-miniprogram/kbone-template-preact)) exist that make the
+onboarding experience for web developers coming from these frameworks easier.
+
+A new project can be created with the `kbone-cli` tool. A wizard asks what framework to initiate the
+project with. The code snippet below shows the Preact demo. In the code snippet below, the `mp`
+command builds the mini app, the `web` command builds the web app, and `build` creates the
+production web app.
+
+```bash
+npx kbone-cli init my-app
+cd my-app
+npm run mp
+npm run web
+npm run build
+```
+
+The code snippet below shows a simple counter component that then gets isomorphically rendered in a
+mini app and a web app. The overhead of the mini app is significant, purely judging from the DOM
+structure.
+
+```js
+import { h, Component } from "preact";
+import "./index.css";
+
+class Counter extends Component {
+  state = { count: 1 };
+
+  sub = () => {
+    this.setState((prevState) => {
+      return { count: --prevState.count };
+    });
+  };
+
+  add = () => {
+    this.setState((prevState) => {
+      return { count: ++prevState.count };
+    });
+  };
+
+  clickHandle = () => {
+    if ("undefined" != typeof wx && wx.getSystemInfoSync) {
+      wx.navigateTo({
+        url: "../log/index?id=1",
+      });
+    } else {
+      location.href = "log.html";
+    }
+  };
+
+  render({}, { count }) {
+    return (
+      <div>
+        <button onClick={this.sub}>-</button>
+        <span>{count}</span>
+        <button onClick={this.add}>+</button>
+        <div onClick={this.clickHandle}>跳转</div>
+      </div>
+    );
+  }
+}
+
+export default Counter;
+```
+
+<figure class="w-figure">
+  <img src="preact-wechat.png" alt="The Preact kbone template demo app opened in WeChat DevTools. Inspecting the DOM structure shows a significant overhead compared to the web app.">
+  <figcaption class="w-figure">
+   The Preact kbone template demo app opened in WeChat DevTools. Note the complex DOM structure and how the plus and minus buttons are actually not <code>&lt;button&gt;</code> elements.
+  </figcaption>
+</figure>
+
+<figure class="w-figure">
+  <img src="preact-web.png" alt="The Preact kbone template demo app opened in the web browser. Inspecting the DOM structure shows the to-be-expected markup based on the Preact component code.">
+  <figcaption class="w-figure">
+   The Preact kbone template demo app opened in the web browser. Note the DOM structure.
+  </figcaption>
+</figure>
+
+## kbone-ui
+
+The [kbone-ui](https://wechat-miniprogram.github.io/kbone/docs/ui/intro/) project
+([open source on GitHub](https://github.com/wechat-miniprogram/kbone-ui)) is a UI framework that
+facilitates both mini app development as well as Vue.js development with kbone. The kbone-ui
+components emulate the look and feel of
+[WeChat's "native" mini app components](https://developers.weixin.qq.com/miniprogram/dev/component/)
```

Comment:
```suggestion
[WeChat's default mini app components](https://developers.weixin.qq.com/miniprogram/dev/component/)
```
Doesn't seem necessary to use. I checked the link and they're not using the word "native"

Reply:
Absolutely.